apps/aecore/lib/aecore/wallet/worker.ex

```diff
@@ -0,0 +1,155 @@
+defmodule Aecore.Wallet.Worker do
+  @defmodule """
+  Module for handling the creation of a Wallet file
+  """
+
+  use GenServer
+
+  alias Aewallet.Wallet
+  alias Aewallet.Encoding
+
+  @typedoc "Wallet type"
+  @type wallet_type :: :ae | :btc
+
+  @typedoc "Options for network"
+  @type opts :: :mainnet | :testnet
+
+  ## Client API
+
+  def start_link(_args) do
+    GenServer.start_link(__MODULE__, %{path: get_aewallet_dir(), pubkey: nil}, name: __MODULE__)
```

Comment:
I think the path shouldn't be stored here since it's already in the env and there is no function to alter it. In general, I think that functions that change the state should be available if a genserver is used.

Reply:
@d-velev can you approve now?